.ipynb_checkpoints/A_Data_collection-checkpoint.ipynb

```diff
@@ -0,0 +1,550 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e7dc4c36",
+   "metadata": {},
+   "source": [
+    "**Using template.py guide created by Roger to create a script that collects Steam reviews via public API**\n",
+    "\n",
+    "Steam Guide: https://partner.steamgames.com/doc/store/getreviews\n",
+    "\n",
+    "Inspired by steamreviews package https://github.com/woctezuma/download-steam-reviews\n",
+    "\n",
+    "API requests limit: 10,000 per day per user https://steamcommunity.com/dev/apiterms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6abff75c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import json\n",
+    "\n",
+    "# Game Steam id\n",
+    "\n",
+    "def download_steam_reviews(game_id,default_cursor):\n",
+    "    \n",
+    "    appid=game_id\n",
+    "    base_url = \"https://store.steampowered.com/appreviews/\"\n",
+    "\n",
+    "    # Parameters for the API request\n",
+    "    params = {\n",
+    "        \"json\": 1, \n",
+    "        \"filter\": \"recent\", \n",
+    "        \"language\": \"english\", \n",
+    "        \"review_type\": \"all\", \n",
+    "        \"purchase_type\": \"all\",\n",
+    "        \"num_per_page\": 100,\n",
+    "    }\n",
+    "\n",
+    "    # Store the reviews\n",
+    "    reviews = dict()\n",
+    "\n",
+    "    cursor = default_cursor\n",
+    "\n",
+    "    # Keep track of how many requests are made\n",
+    "    counter = 0\n",
+    "\n",
+    "    # Keep track of how many reviews are fetched\n",
+    "    reviews_fetched = 0\n",
+    "    \n",
+    "    print(\"Starting to send requests...\")\n",
+    "\n",
+    "    while True:\n",
+    "        counter += 1\n",
+    "\n",
+    "        # Append the cursor to the parameters\n",
+    "        params[\"cursor\"] = cursor\n",
+    "\n",
+    "        try:\n",
+    "            response = requests.get(base_url + str(appid), params=params)\n",
+    "\n",
+    "            # Check if the request was successful\n",
+    "            if response.status_code == 200:\n",
+    "\n",
+    "                data = response.json()\n",
+    "\n",
+    "                # Check if the query was successful\n",
+    "                if data[\"success\"] == 1:\n",
+    "                    # Get user reviews\n",
+    "                    reviews_list = data[\"reviews\"]\n",
+    "\n",
+    "                    for review in reviews_list:\n",
+    "\n",
+    "                        recommendationid = review[\"recommendationid\"]\n",
+    "                        review_data = review\n",
+    "\n",
+    "                        reviews[recommendationid] = review_data\n",
```

Comment:
superfluous declaration of review_data since it's only used for passing the value to the dict.